test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -728,6 +779,46 @@ public void CreateDisplayMetadata_IsFlagsEnum_ReflectsModelType(Type type, bool
         }
```

Comment:
`CreateDisplayMetadata_EnumGroupedDisplayNamesAndValues_ReflectsModelType` fails with our new EnumGroupAndName.Equals because all the EnumGroupAndName's  in the MemberData don't use `FieldInfo` (and thus _displayattribute). Would it be enough for this test to check that the Name and group values on `EnumGroupedDisplayNamesAndValues` are what they should be? Because changing the MemberData to use the FieldInfo's would be basically testing that it equals itself, correct?